commit d00dc0d48c1f4ed82ebdb7df212607f899e9577d
Author: Doris Tam <doristam@catalyst.net.nz>
Date:   Wed Jun 12 11:07:49 2024 +1200

    upgrade steps from react native 0.72 to 0.73

    okay to install

    these package-lock on legacy-peer-deps keep changing

    Fixing:..

    Finally building!!!

diff --git a/MaharaMobile/package.json b/MaharaMobile/package.json
index 11dee979bc..cc0c42c68f 100644
--- a/MaharaMobile/package.json
+++ b/MaharaMobile/package.json
@@ -20,83 +20,84 @@
     "postversion": "react-native-version",
     "devices": "adb devices",
     "check-deps": "rnx-dep-check",
-    "fix-deps": "rnx-dep-check --write"
+    "fix-deps": "rnx-dep-check --write",
+    "clean": "npx react-native-clean-project"
   },
   "codegenConfig": {
     "name": "MaharaMobile",
     "type": "components",
     "jsSrcsDir": "./src",
     "android": {
-      "javaPackageName": "org.mahara.mobile.app"
+      "javaPackageName": "org.mahara.mobile.app"
     }
   },
   "dependencies": {
-    "@fortawesome/free-regular-svg-icons": "^6.4.2",
-    "@fortawesome/free-solid-svg-icons": "^6.4.2",
-    "@gluestack-style/react": "^1.0.47",
-    "@gluestack-ui/themed-native-base": "^1.1.4",
-    "@lingui/core": "^4.4.0",
-    "@lingui/react": "^4.4.0",
-    "@react-native-async-storage/async-storage": "^1.19.2",
+    "@babel/preset-typescript": "^7.24.7",
+    "@fortawesome/free-regular-svg-icons": "^6.5.2",
+    "@fortawesome/free-solid-svg-icons": "^6.5.2",
+    "@gluestack-style/react": "^1.0.56",
+    "@gluestack-ui/config": "^1.1.18",
+    "@gluestack-ui/themed-native-base": "^1.1.29",
+    "@lingui/core": "^4.11.1",
+    "@lingui/react": "^4.11.1",
+    "@react-native-async-storage/async-storage": "^1.23.1",
     "@react-native-community/masked-view": "^0.1.11",
-    "@react-navigation/bottom-tabs": "^6.5.8",
-    "@react-navigation/native": "^6.1.7",
-    "@react-navigation/stack": "^6.3.17",
-    "@reduxjs/toolkit": "^1.9.5",
-    "make-plural": "^7.3.0",
-    "native-base": "^3.4.28",
+    "@react-navigation/bottom-tabs": "^6.5.20",
+    "@react-navigation/native": "^6.1.17",
+    "@react-navigation/stack": "^6.3.29",
+    "@reduxjs/toolkit": "^2.2.5",
+    "make-plural": "^7.4.0",
     "react": "18.2.0",
-    "react-native": "0.72.4",
-    "react-native-blob-util": "^0.19.1",
-    "react-native-device-info": "^10.8.0",
-    "react-native-document-picker": "^9.0.1",
-    "react-native-gesture-handler": "^2.12.1",
-    "react-native-image-picker": "^5.6.1",
-    "react-native-localize": "^3.0.2",
-    "react-native-permissions": "^3.8.4",
-    "react-native-safe-area-context": "^4.7.1",
-    "react-native-screens": "^3.24.0",
-    "react-native-svg": "^13.4.0",
-    "react-native-uuid": "^2.0.1",
+    "react-native": "0.73.8",
+    "react-native-blob-util": "^0.19.9",
+    "react-native-device-info": "^11.1.0",
+    "react-native-document-picker": "^9.3.0",
+    "react-native-gesture-handler": "^2.16.2",
+    "react-native-image-picker": "^7.1.2",
+    "react-native-localize": "^3.2.0",
+    "react-native-permissions": "^4.1.5",
+    "react-native-safe-area-context": "^4.10.4",
+    "react-native-screens": "^3.31.1",
+    "react-native-svg": "^15.3.0",
+    "react-native-uuid": "^2.0.2",
     "react-native-version": "^4.0.0",
-    "react-native-webview": "^13.3.1",
-    "react-redux": "^8.1.2",
+    "react-native-webview": "^13.10.3",
+    "react-redux": "^9.1.2",
     "redux-logger": "^3.0.6"
   },
   "devDependencies": {
-    "@babel/core": "^7.22.10",
+    "@babel/core": "^7.20.0",
     "@babel/preset-env": "^7.20.0",
-    "@babel/preset-typescript": "^7.22.15",
     "@babel/runtime": "^7.20.0",
-    "@fortawesome/react-native-fontawesome": "^0.3.0",
-    "@lingui/cli": "^4.4.0",
-    "@lingui/macro": "^4.4.0",
+    "@fortawesome/react-native-fontawesome": "^0.3.2",
+    "@lingui/cli": "^4.11.1",
+    "@lingui/macro": "^4.11.1",
     "@react-native-community/eslint-config": "^3.2.0",
-    "@react-native/eslint-config": "^0.72.2",
-    "@react-native/metro-config": "^0.72.11",
-    "@testing-library/react-native": "^12.3.0",
-    "@trivago/prettier-plugin-sort-imports": "^4.2.0",
-    "@tsconfig/react-native": "^3.0.0",
-    "@types/jest": "^29.5.4",
-    "@types/react": "^18.0.24",
+    "@react-native/babel-preset": "0.73.21",
+    "@react-native/eslint-config": "0.73.2",
+    "@react-native/metro-config": "0.73.5",
+    "@react-native/typescript-config": "0.73.1",
+    "@testing-library/react-native": "^12.5.1",
+    "@trivago/prettier-plugin-sort-imports": "^4.3.0",
+    "@tsconfig/react-native": "^3.0.5",
+    "@types/jest": "^29.5.12",
+    "@types/react": "^18.3.3",
     "@types/react-test-renderer": "^18.0.0",
-    "babel-jest": "^29.2.1",
+    "babel-jest": "^29.6.3",
     "babel-plugin-macros": "^3.1.0",
-    "babel-plugin-module-resolver": "^5.0.0",
+    "babel-plugin-module-resolver": "^5.0.2",
     "eslint": "^8.19.0",
     "eslint-config-airbnb": "^19.0.4",
-    "eslint-config-airbnb-typescript": "^17.1.0",
-    "eslint-config-prettier": "^9.0.0",
-    "eslint-plugin-react-native": "^4.0.0",
-    "jest": "^29.6.4",
-    "metro-react-native-babel-preset": "0.76.8",
-    "prettier": "^2.4.1",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-plugin-react-native": "^4.1.0",
+    "jest": "^29.6.3",
+    "prettier": "2.8.8",
     "react-test-renderer": "18.2.0",
     "redux-mock-store": "^1.5.4",
-    "ts-jest": "^29.1.1",
-    "typescript": "^4.8.4"
+    "ts-jest": "^29.1.4",
+    "typescript": "5.0.4"
   },
   "engines": {
-    "node": ">=16"
+    "node": ">=18"
   }
 }
